spec/rubocop/cop/lint/ordered_magic_comments_spec.rb

```diff
@@ -0,0 +1,80 @@
+# frozen_string_literal: true
+
+RSpec.describe RuboCop::Cop::Lint::OrderedMagicComments, :config do
+  subject(:cop) { described_class.new(config) }
+
+  it 'registers an offense when using `frozen_string_literal` magic comment ' \
+     'is first line' do
+    expect_offense(<<-RUBY.strip_indent)
+      # frozen_string_literal: true
+      # coding: ascii
```

Comment:
I guess you should also add some test for `encoding`, which is also valid.

Reply:
Thank you for review. I made the following changes by 99180fe.

- Improve description and offense message about this cop.
- Change `coding: ascii` to `encoding: ascii` and added some tests of `coding: ascii` and `-*- encoding : ascii-8bit -*-`.